tcms/rpc/tests/test_testrun.py

```diff
@@ -251,6 +301,48 @@ def test_create_with_product_version(self):
         self.assertEqual(result['product_version'], self.plan.product_version.value)
 
 
+class TestFilter(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.plan = TestPlanFactory()
+
+        self.test_case = TestCaseFactory()
+        self.test_case.save()
+        self.plan.add_case(self.test_case)
+
+        self.test_run = TestRunFactory(plan=self.plan)
+
+    def test_empty_query(self):
+        result = self.rpc_client.TestRun.filter()
+        self.assertTrue(isinstance(result, list))
+        self.assertEqual(1, len(result))
+        self.assertEqual(self.test_run.pk, result[0]['id'])
+
+    def test_filter(self):
+        result = self.rpc_client.TestRun.filter({'plan': self.plan.pk, 'stop_date__isnull': True})
+        self.assertEqual(1, len(result))
+        self.assertEqual(self.test_run.pk, result[0]['id'])
+
+
+class TestFilterPermission(APIPermissionsTestCase):
+    permission_label = 'testruns.view_testrun'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.test_run = TestRunFactory()
+
+    def verify_api_with_permission(self):
+        result = self.rpc_client.TestRun.filter({
+            'summary__icontains': "test run for", 'stop_date__isnull': True})
```

Comment:
because the value is hard coded and may not match what's in the DB this is essentially the same as doing an empty filter to fetch all objects. If you drop the setup method it should still work (in order to validate the permissions).